lib/api/core/models/repositories/repository.js

```diff
@@ -97,15 +103,19 @@ class Repository {
 
           return this.fromDTO(dto);
         }
+
         return null;
       })
       .catch(error => {
-        if (error.status === 404) {
-          // no content found, we return null without failing
-          return Bluebird.resolve(null);
+        if (error.status !== 404) {
+          return Bluebird.reject(error);
```

Comment:
(nitpicking) You can use `throw` instead of `return Bluebird.reject`
That remark can also be applied to the other rejection just below